packages/data-stores/src/subscriber/actions.ts

```diff
@@ -71,15 +71,19 @@ export function createActions() {
 					[ 'parse_only', String( parseOnly ) ] as [ string, string ],
 				];
 
-				const data: ImportSubscribersResponse = await wpcomProxyRequest( {
+				const data = ( await wpcomProxyRequest( {
 					path: `/sites/${ encodeURIComponent( siteId ) }/subscribers/import`,
 					method: 'POST',
 					apiNamespace: 'wpcom/v2',
 					token: typeof token === 'string' ? token : undefined,
 					formData: formDataEntries as ( string | File )[][],
-				} );
+				} ) ) as ImportSubscribersResponse & { error?: { code: string; message: string } };
 
-				dispatch.importCsvSubscribersStartSuccess( siteId, data.upload_id );
+				if ( data.upload_id ) {
```

Comment:
This is the bit where we are handling the respone from WPCOM. The else part takes care of the WP_Errors that are returned with a 200 error code.